src/VisualStudio/Core/Impl/SolutionExplorer/SourceGeneratedFileItems/SourceGeneratedFileItemSourceProvider.cs

```diff
@@ -12,35 +12,23 @@
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Utilities;
 
-namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer
-{
-    [Export(typeof(IAttachedCollectionSourceProvider))]
-    [Name(nameof(SourceGeneratedFileItemSourceProvider))]
-    [Order]
-    [AppliesToProject("CSharp | VB")]
-    internal sealed class SourceGeneratedFileItemSourceProvider : AttachedCollectionSourceProvider<SourceGeneratorItem>
-    {
-        private readonly Workspace _workspace;
-        private readonly IAsynchronousOperationListener _asyncListener;
-        private readonly IThreadingContext _threadingContext;
-
-        [ImportingConstructor]
-        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-        public SourceGeneratedFileItemSourceProvider(VisualStudioWorkspace workspace, IAsynchronousOperationListenerProvider asyncListenerProvider, IThreadingContext threadingContext)
-        {
-            _workspace = workspace;
-            _asyncListener = asyncListenerProvider.GetListener(FeatureAttribute.SourceGenerators);
-            _threadingContext = threadingContext;
-        }
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer;
 
-        protected override IAttachedCollectionSource? CreateCollectionSource(SourceGeneratorItem item, string relationshipName)
-        {
-            if (relationshipName == KnownRelationships.Contains)
-            {
-                return new SourceGeneratedFileItemSource(item, _workspace, _asyncListener, _threadingContext);
-            }
+[Export(typeof(IAttachedCollectionSourceProvider))]
+[Name(nameof(SourceGeneratedFileItemSourceProvider)), Order]
```

Comment:
>Order [](http://example.com/codeflow?start=54&length=5)

Definitely a nit, but I'm curious why this is preferred. I assume it's because there is a relation between this attributes, but it seems easier to miss the OrderAttribute in this form.

Reply:
yes.  that's basically it.  (imo, they should have been merge din the first place :)).